src/node.cpp

```diff
@@ -1540,6 +1535,11 @@ LocalNode::~LocalNode()
         return;
     }
 
+    if (!sync->client)
```

Comment:
Here and in other places we have to check for `client` being null because it is null in unit tests.